### PR TITLE
Spanish Patient Part 1

### LINKED ISSUE #1571: I18n Spanish: Patients - Part 1 (of 3)
## What are we trying to do?

This ticket is part of the DARIA translation effort. The goal is to implement all our text in both English and Spanish through the i18n* process. 

This ticket is for the Patient views.  
Code links:  
app/views/patients/edit.html.erb  
app/views/patients/_menu.html.erb  
app/views/patients/_patient_information.html.erb  

The current English text should be moved to `config/locales/en.yml`,   
and new Spanish text should be added to `config/locales/es.yml`.

**Google translate is perfectly acceptable!**  
PRs will be reviewed by a fluent CM to correct any weirdness.

### Steps

For each view  
 - find each instance of displayed text on the view
   `<%= f.date_field :appointment_date, label: 'Appointment date', autocomplete: 'off', help: "Approx gestation at appt: #{patient.last_menstrual_period_at_appt}" %>`
 - replace it with a call to i18n `t` function with a unique identifier based on the file path & field
   `<%= f.date_field :appointment_date, label: t('patient.dashboard.appt_date'), autocomplete: 'off', help: t('patient.dashboard.approx_gestation', gest: patient.last_menstrual_period_at_appt) %>`
 - add a config entry for each language with that identifier as the key, and the language line as the value.
   `en.yml`:
   ```
   patient:
     dashboard:
       approx_gestation: 'Approx gestation at appt: %{gest}'
       appt_date: Appointment date
    ```  
    `es.yml`:
    ```
    patient:
      dashboard:
        approx_gestation: 'Aprox la gestación en la cita: %{gest}'
        appt_date: Día de la cita
    ```

### Completed View Example

For an example of a completed view translation, see [the footer view i18n PR](https://github.com/DCAFEngineering/dcaf_case_management/pull/1547/files).

### Formatting in the YAML files

Each view should have a it's own section. If you see the same key being used across Some commonly used values can go under a shared secton. `app/views/patients/_patient_dashboard.html.erb` keys go under `patient:` -> `dashboard:`.

### Finalizing with i18n-tasks

After you finish all the keys, you should run the i18n checker before committing! This sorts the keys in a standard way and checks for missing keys.

`i18n-tasks normalize`
`i18n-tasks health` - address any issues this highlights!

[More info on using i18n-tasks](https://github.com/glebm/i18n-tasks)

\* 'i18n' is shorthand for 'Internationalization'. [More info on i18n in Rails](https://guides.rubyonrails.org/i18n.html)

Part of #1382

Patient 2/3: #1572 
Patient 3/3: #1573

### PR DESCRIPTION
**Pull request by lomky**
I rule and have completed some work on Case Manager that's ready for review!

Working through the views in #1571 

This pull request makes the following changes:
* Base labels & strings translated
* Helper text
* Dropdowns

screenshot_2019-02-15 daria

It relates to the following issue #s: 
* Fixes #1571
* Bumps #1382

### CODE REVIEW

**Comment by lomky** [comment]
@montanezp for review! I'd appreciate an extra level dose of skepticism on my translations here. I'm suspicious of some of Google's offers, and this is a pretty critical view! Don't hold back :muscle: 

Still a bit more to come, but feel free to dig in!

**Comment by lomky** [comment]
@colinxfleming the only thing missing now are how to handle dropdown options added via Config.

In some cases these are untranslatable, as we'll never know what fields folks are entering. But I'm pondering if there's a way to handle obvious ones (i.e. 'Spanish' for a language option). Or am I now entering over-engineering?

Thoughts?

File: config/locales/en.yml [hunk 1]
```diff
@@ -87,9 +87,93 @@ en:
...
+      other_contact:
+        name: Other contact name
+        phone: Other phone
+        relationship: Relationship to other contact
+        title: Other Contact
+      race_eth: Race / Ethnicity
+      referred_by: How patient heard about %{fund}
+      special_circumstances:
+        domestic_violence: Domestic violence
+        fetal: Fetal diagnosis
```
Comment (colinxfleming) [inline config/locales/en.yml]:
```suggestion
        fetal_diagnosis: Fetal diagnosis
```

File: config/locales/en.yml [hunk 2]
```diff
@@ -87,9 +87,93 @@ en:
...
+      city: City
+      county: County
+      employ: Employment Status
+      household_adults: Adults in household
+      household_help: "(including patient)"
+      household_minors: Minors in household
+      income: Income
+      income_help: Employment, Food Stamps, SS, TANF, etc.
+      insurance: Patient insurance
+      lang: Preferred Language
```
Comment (colinxfleming) [inline config/locales/en.yml]:
```suggestion
      language: Preferred Language
```

File: config/locales/en.yml [hunk 3]
```diff
@@ -87,9 +87,93 @@ en:
...
+        friend: Friend
+        fund: "%{fund} website or social media"
+        homeless: Homeless shelter
+        legal: Legal clinic
+        naf: NAF
+        nnaf: NNAF
+        other_fund: Other abortion fund
+        prev_patient: Previous patient
+        school: School
+        search: Google/Web search
```
Comment (colinxfleming) [inline config/locales/en.yml]:
```suggestion
        web_search: Google/Web search
```

File: config/locales/en.yml [hunk 4]
```diff
@@ -87,9 +87,93 @@ en:
...
+      refer:
+        clinic: Clinic
+        crisis_org: Sexual assault crisis org
+        cvac: Crime victim advocacy center
+        dv_org: Domestic violence crisis/intervention org
+        family: Family member
+        friend: Friend
+        fund: "%{fund} website or social media"
+        homeless: Homeless shelter
+        legal: Legal clinic
```
Comment (colinxfleming) [inline config/locales/en.yml]:
```suggestion
        legal_clinic: Legal clinic
```

File: config/locales/en.yml [hunk 5]
```diff
@@ -87,9 +87,93 @@ en:
...
+        black_afican_american: Black/African-American
+        hispanic_latino: Hispanic/Latino
+        mixed_race_ethnicity: Mixed Race/Ethnicity
+        native_american: Native American
+        native_hawaiian_pacific_islander: Native Hawaiian or Pacific Islander
+        other: Other
+        white_caucasian: White/Caucasian
+      refer:
+        clinic: Clinic
+        crisis_org: Sexual assault crisis org
```
Comment (colinxfleming) [inline config/locales/en.yml]:
```suggestion
        sexual_assault_crisis_org: Sexual assault crisis org
```

**Comment by lomky** [comment]
Much :green_heart: for calling out my bad key name habits! I'll make them clearer :)

**Comment by lomky** [comment]
@montanezp @colinxfleming Here are a couple shots of the Config fields. I'm thinking the easiest technical fix is to update the Config creation process to enter those values bilingually.

screenshot from 2019-02-16 17-23-23

(ignore the line down the center, it's just two screenshots lined up)

File: config/locales/es.yml [hunk 6]
```diff
@@ -90,6 +90,37 @@ es:
       week:
         one: "%{count} semana"
         other: "%{count} semanas"
+    information:
+      age: Años
+      city: Ciudad
+      county: Condado
+      employ: Estado de Empleo
```
Comment (montanezp) [inline config/locales/es.yml]: Estatus de Empleo

File: config/locales/es.yml [hunk 7]
```diff
@@ -90,6 +90,37 @@ es:
...
+      household_adults: Adultos en el Hogar
+      household_minors: Menores en el Hogar
+      income: Ingresos
+      insurance: Seguro del Paciente
```
Comment (montanezp) [inline config/locales/es.yml]: Seguro Médico del Paciente

File: config/locales/es.yml [hunk 8]
```diff
@@ -90,6 +90,37 @@ es:
...
+        domestic_violence: Violencia doméstica
+        fetal: Diagnostico fetal
+        homelessness: La falta de vivienda
+        incest: Incesto
+        other_med: Otro problema médico
+        prison: Prisión
+        rape: Violación
+        title: Circunstancias Especiales
+      state: Estado
+      textable: "¿Textable?"
```
Comment (montanezp) [inline config/locales/es.yml]: Se permite enviar mensajes de texto?

File: config/locales/es.yml [hunk 9]
```diff
@@ -90,6 +90,37 @@ es:
...
+      title: Información del Paciente
+      voicemail: Preferencias de Mensaje de Voz
     shared:
       name: Nombre y apellido
       status: 'Estado:'
```
Comment (montanezp) [inline config/locales/es.yml]: Estatus

File: config/locales/es.yml [hunk 10]
```diff
@@ -87,26 +87,70 @@ es:
       day:
         one: "%{count} día"
         other: "%{count} días"
+      employ:
+        full: Tiempo completo
+        odd: Trabajos ocasionales
+        part: Medio tiempo
+        student: Estudiante
+        unemp: Desempleados
```
Comment (montanezp) [inline config/locales/es.yml]: Desempleada

File: config/locales/es.yml [hunk 11]
```diff
@@ -87,26 +87,70 @@ es:
...
+        family: Miembro de la familia
+        friend: Amigo
+        fund: "%{fund} sitio web o redes sociales"
+        homeless: Refugio para indigentes
+        legal: Clínica legal
+        naf: NAF
+        nnaf: NNAF
+        other_fund: Otro fondo de aborto
+        prev_patient: Paciente anterior
+        school: Colegio
```
Comment (montanezp) [inline config/locales/es.yml]: escuela

**Comment by rudietuesdays** [comment]
@lomky apologies for hoping on this late, i know you've already merged but it also looks great from my end! Just also flagging about translating the sidebar text as well ("Patient Information," "Abortion Information," ... "Cancel Pledge") -- is that covered in a different ticket?

**Pull request merged**